From 176e1bfabe6ce1935c80e875550612801b75b3e5 Mon Sep 17 00:00:00 2001
From: Aymen Zayet <aymen.zayet@intel.com>
Date: Thu, 27 Dec 2012 17:40:55 +0100
Subject: [PORT FROM MAIN] wl12xx: reduce tx_watchdog_timeout to 2 secs

BZ: 77349

tx_watchdog_timeout is used to detect data stall on tx buffer.
This may happen when the firmware hangs (on BT coex use cases).
The default value of 5 secs is too high. As a consequence, SIP
session facing this issue will hangup and the call is terminated
unexpectedly.

By reducing tx_watchdog_timeout, we can prevent such situation,
and the wifi firmware recovers properly.

Change-Id: Id68d937c58037ab1af7566210993861ca9a80a98
Orig-Change-Id: I3fc76f0e86cf0b2d00d367e212bc15014960924d
Signed-off-by: Aymen Zayet <aymen.zayet@intel.com>
Reviewed-on: http://android.intel.com:8080/84090
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index a29ae60..98406ed 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -217,7 +217,7 @@ static struct conf_drv_settings default_conf = {
 		.basic_rate_5                = CONF_HW_BIT_RATE_6MBPS,
 		.tmpl_short_retry_limit      = 10,
 		.tmpl_long_retry_limit       = 10,
-		.tx_watchdog_timeout         = 5000,
+		.tx_watchdog_timeout         = 2000,
 	},
 	.conn = {
 		.wake_up_event               = CONF_WAKE_UP_EVENT_DTIM,
-- 
1.7.4.1

